FTBFS: utsushi

```
  CXXLD    verify
make[3]: Leaving directory '/home/build/YPKG/root/utsushi/build/imagescan-3.65.0/drivers/esci'
Making all in tests
make[3]: Entering directory '/home/build/YPKG/root/utsushi/build/imagescan-3.65.0/drivers/esci/tests'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/build/YPKG/root/utsushi/build/imagescan-3.65.0/drivers/esci/tests'
make[2]: Leaving directory '/home/build/YPKG/root/utsushi/build/imagescan-3.65.0/drivers/esci'
Making all in sane
make[2]: Entering directory '/home/build/YPKG/root/utsushi/build/imagescan-3.65.0/sane'
Making all in .
make[3]: Entering directory '/home/build/YPKG/root/utsushi/build/imagescan-3.65.0/sane'
  CXX      backend.lo
In file included from backend.cpp:56:
version.hpp:41:2: error: #error "SANE installation violates versioning portability constraints."
   41 | #error "SANE installation violates versioning portability constraints."
      |  ^~~~~
```

Package is archived upstream so a deprecation may be suitable. `libboost` rebuilds are blocked on this package.

--- a/packages/u/utsushi/abi_used_libs
+++ b/packages/u/utsushi/abi_used_libs
@@ -7,6 +7,7 @@ libboost_program_options.so.1.83.0
 libc.so.6
 libgcc_s.so.1
 libgdkmm-2.4.so.1
+libgiomm-2.4.so.1
 libglib-2.0.so.0
 libglibmm-2.4.so.1
 libgobject-2.0.so.0

--- a/packages/u/utsushi/abi_used_symbols
+++ b/packages/u/utsushi/abi_used_symbols
@@ -1,3 +1,8 @@
+UNKNOWN:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE10_M_disposeEv
+UNKNOWN:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE4swapERS4_
+UNKNOWN:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE7reserveEm
+UNKNOWN:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE9_M_assignERKS4_
+UNKNOWN:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE9_M_mutateEmmPKcm
 UNKNOWN:_ZTIN3Atk11ImplementorE
 UNKNOWN:_ZTIN3Gtk10CellLayoutE
 UNKNOWN:_ZTIN3Gtk11ActionGroupE
@@ -709,7 +714,6 @@ libstdc++.so.6:_ZNSo9_M_insertIbEERSoT_
 libstdc++.so.6:_ZNSo9_M_insertIdEERSoT_
 libstdc++.so.6:_ZNSo9_M_insertIlEERSoT_
 libstdc++.so.6:_ZNSo9_M_insertImEERSoT_
-libstdc++.so.6:_ZNSoC2Ev
 libstdc++.so.6:_ZNSolsEi
 libstdc++.so.6:_ZNSolsEs
 libstdc++.so.6:_ZNSt11logic_errorC1EPKc
@@ -758,7 +762,6 @@ libstdc++.so.6:_ZNSt15basic_streambufIcSt11char_traitsIcEE8overflowEi
 libstdc++.so.6:_ZNSt15basic_streambufIcSt11char_traitsIcEE9pbackfailEi
 libstdc++.so.6:_ZNSt15basic_streambufIcSt11char_traitsIcEE9showmanycEv
 libstdc++.so.6:_ZNSt15basic_streambufIcSt11char_traitsIcEE9underflowEv
-libstdc++.so.6:_ZNSt15basic_streambufIcSt11char_traitsIcEEC2Ev
 libstdc++.so.6:_ZNSt16invalid_argumentC1EPKc
 libstdc++.so.6:_ZNSt16invalid_argumentC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
 libstdc++.so.6:_ZNSt16invalid_argumentD1Ev
@@ -792,7 +795,6 @@ libstdc++.so.6:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE4swapERS4_
 libstdc++.so.6:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE6appendEPKc
 libstdc++.so.6:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE6appendEPKcm
 libstdc++.so.6:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE6resizeEmc
-libstdc++.so.6:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE7replaceEmmPKcm
 libstdc++.so.6:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE7reserveEm
 libstdc++.so.6:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE8_M_eraseEmm
 libstdc++.so.6:_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE9_M_appendEPKcm
@@ -835,7 +837,6 @@ libstdc++.so.6:_ZNSt9bad_allocD2Ev
 libstdc++.so.6:_ZNSt9basic_iosIcSt11char_traitsIcEE4initEPSt15basic_streambufIcS1_E
 libstdc++.so.6:_ZNSt9basic_iosIcSt11char_traitsIcEE5clearESt12_Ios_Iostate
 libstdc++.so.6:_ZNSt9basic_iosIcSt11char_traitsIcEE5imbueERKSt6locale
-libstdc++.so.6:_ZNSt9basic_iosIcSt11char_traitsIcEEC2Ev
 libstdc++.so.6:_ZNSt9exceptionD2Ev
 libstdc++.so.6:_ZSt16__ostream_insertIcSt11char_traitsIcEERSt13basic_ostreamIT_T0_ES6_PKS3_l
 libstdc++.so.6:_ZSt16__throw_bad_castv
@@ -890,7 +891,6 @@ libstdc++.so.6:_ZTISt9exception
 libstdc++.so.6:_ZTIb
 libstdc++.so.6:_ZTId
 libstdc++.so.6:_ZTIi
-libstdc++.so.6:_ZTIm
 libstdc++.so.6:_ZTIv
 libstdc++.so.6:_ZTTNSt7__cxx1118basic_stringstreamIcSt11char_traitsIcESaIcEEE
 libstdc++.so.6:_ZTTNSt7__cxx1119basic_istringstreamIcSt11char_traitsIcESaIcEEE
@@ -929,6 +929,7 @@ libstdc++.so.6:__cxa_allocate_exception
 libstdc++.so.6:__cxa_bad_cast
 libstdc++.so.6:__cxa_bad_typeid
 libstdc++.so.6:__cxa_begin_catch
+libstdc++.so.6:__cxa_call_terminate
 libstdc++.so.6:__cxa_demangle
 libstdc++.so.6:__cxa_end_catch
 libstdc++.so.6:__cxa_free_exception

--- a/packages/u/utsushi/files/0001-Remove-constraint-on-sane-minor-version-0.patch
+++ b/packages/u/utsushi/files/0001-Remove-constraint-on-sane-minor-version-0.patch
@@ -1,0 +1,23 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Thomas Staudinger <Staudi.Kaos@gmail.com>
+Date: Thu, 12 Sep 2024 10:31:27 +0200
+Subject: [PATCH] Remove constraint on sane minor version "0"
+
+Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
+---
+ sane/version.hpp | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/sane/version.hpp b/sane/version.hpp
+index 11b5f29..512855d 100644
+--- a/sane/version.hpp
++++ b/sane/version.hpp
+@@ -37,7 +37,7 @@ extern "C" {                    // needed until sane-backends-1.0.14
+ #include <sane/sane.h>
+ }
+ 
+-#if !((SANE_MAJOR == SANE_CURRENT_MAJOR) && (SANE_MINOR == 0))
++#if !(SANE_MAJOR == SANE_CURRENT_MAJOR)
+ #error "SANE installation violates versioning portability constraints."
+ #endif
+ 

--- a/packages/u/utsushi/package.yml
+++ b/packages/u/utsushi/package.yml
@@ -1,6 +1,6 @@
 name       : utsushi
 version    : 3.65.0
-release    : 26
+release    : 27
 source     :
     - https://gitlab.com/utsushi/imagescan/-/archive/3.65.0/imagescan-3.65.0.tar.gz : 8bc301c21ee0c2cf65e871a0309d21372a46528669ffc6abcfe8d9d5fff94e14
 homepage   : https://gitlab.com/utsushi/imagescan
@@ -19,6 +19,8 @@ builddeps  :
     - libboost-devel
 setup      : |
     %patch -p1 -i $pkgfiles/boost-1.74.patch
+    # Fix build with sane-backends > 1.0.x
+    %patch -p1 -i $pkgfiles/0001-Remove-constraint-on-sane-minor-version-0.patch
     %configure_no_runstatedir \
                 --with-udev-confdir=/usr/lib64/udev \
                 --enable-sane-config \

--- a/packages/u/utsushi/pspec_x86_64.xml
+++ b/packages/u/utsushi/pspec_x86_64.xml
@@ -3,8 +3,8 @@
         <Name>utsushi</Name>
         <Homepage>https://gitlab.com/utsushi/imagescan</Homepage>
         <Packager>
-            <Name>Joey Riches</Name>
-            <Email>josephriches@gmail.com</Email>
+            <Name>Thomas Staudinger</Name>
+            <Email>Staudi.Kaos@gmail.com</Email>
         </Packager>
         <License>GPL-3.0-or-later</License>
         <PartOf>office</PartOf>
@@ -26,7 +26,7 @@
             <Path fileType="library">/usr/lib64/sane/libsane-utsushi.la</Path>
             <Path fileType="library">/usr/lib64/sane/libsane-utsushi.so</Path>
             <Path fileType="library">/usr/lib64/sane/libsane-utsushi.so.1</Path>
-            <Path fileType="library">/usr/lib64/sane/libsane-utsushi.so.1.0.30</Path>
+            <Path fileType="library">/usr/lib64/sane/libsane-utsushi.so.1.3.1</Path>
             <Path fileType="library">/usr/lib64/udev/rules.d/utsushi-esci.rules</Path>
             <Path fileType="library">/usr/lib64/utsushi/libcnx-hexdump.so</Path>
             <Path fileType="library">/usr/lib64/utsushi/libcnx-hexdump.so.0</Path>
@@ -51,7 +51,7 @@
             <Path fileType="library">/usr/lib64/utsushi/libutsushi.so.0.0.0</Path>
             <Path fileType="library">/usr/lib64/utsushi/sane/libsane-utsushi.so</Path>
             <Path fileType="library">/usr/lib64/utsushi/sane/libsane-utsushi.so.1</Path>
-            <Path fileType="library">/usr/lib64/utsushi/sane/libsane-utsushi.so.1.0.30</Path>
+            <Path fileType="library">/usr/lib64/utsushi/sane/libsane-utsushi.so.1.3.1</Path>
             <Path fileType="library">/usr/lib64/utsushi/utsushi/doc-locate</Path>
             <Path fileType="library">/usr/lib64/utsushi/utsushi/esci-interpreter</Path>
             <Path fileType="library">/usr/lib64/utsushi/utsushi/get-text-orientation</Path>
@@ -122,7 +122,7 @@
 </Description>
         <PartOf>programming.devel</PartOf>
         <RuntimeDependencies>
-            <Dependency release="26">utsushi</Dependency>
+            <Dependency release="27">utsushi</Dependency>
         </RuntimeDependencies>
         <Files>
             <Path fileType="header">/usr/include/utsushi/array.hpp</Path>
@@ -169,12 +169,12 @@
         </Files>
     </Package>
     <History>
-        <Update release="26">
-            <Date>2023-11-28</Date>
+        <Update release="27">
+            <Date>2024-09-12</Date>
             <Version>3.65.0</Version>
             <Comment>Packaging update</Comment>
-            <Name>Joey Riches</Name>
-            <Email>josephriches@gmail.com</Email>
+            <Name>Thomas Staudinger</Name>
+            <Email>Staudi.Kaos@gmail.com</Email>
         </Update>
     </History>
 </PISI>

